### DISCUSSION TITLE
WalletConnectConnector loses connection on page refresh

### DISCUSSION THREAD

**Discussion by newyokai** [wevm/wagmi#1503] [General]
Hi all, I created a WalletConnect connector and I have a problem in page refresh. Every time I do a refresh, the useAccount() hook tells me that no wallet is connected. This always happens only if I refresh the pag. Any advice? 
`const { account, chain } = await connectAsync({ connector: new WalletConnectConnector({
            options: {
              qrcode:true,
              rpc: {
                5:"..."
              }
            }
          }) });`

**Comment by tmm** [comment]
What does you `Client` instance look like? The wagmi `Client` only autoconnects to [connectors attached to it](https://wagmi.sh/react/client#connectors-optional). If you add the `WalletConnectConnector` to `createClient({ connectors: […] })` it should work.

**Reply by newyokai** [reply]
const { chains, provider } = configureChains(
  [goerli],
  [
    jsonRpcProvider({
      rpc: (goerli) => ({
        http: `...`,
      }),
    }),
  ],
)
const client = createClient({
  provider,
  autoConnect:true,
});


This is my client instance. If the user authenticate via MetamaskConnector it works also on refresh, with WalletConnect no...

**Reply by tmm** [reply]
You need to add `WalletConnectConnector` to `connector` in `createClient`, otherwise the wagmi `Client` will not have access to this when attempting to autoconnect. (You should probably also add `MetamaskConnector` too. It just so happens to work because the browser injected wallet exists across page reloads.)

**Reply by newyokai** [reply]
Man you saved me! Really thank you, it works!
